src/main/java/org/spongepowered/common/data/processor/multi/entity/InvisibilityDataProcessor.java

```diff
@@ -67,13 +67,9 @@ protected boolean set(final VanishableBridge dataHolder, final Map<Key<?>, Objec
         final boolean untargetable = (Boolean) keyValues.get(Keys.VANISH_PREVENTS_TARGETING);
         final boolean vanish = (Boolean) keyValues.get(Keys.VANISH);
         dataHolder.bridge$setInvisible(invis);
-        if (vanish) {
-            dataHolder.bridge$setVanished(true);
-            dataHolder.bridge$setUncollideable(collision);
-            dataHolder.bridge$setUntargetable(untargetable);
-        } else {
-            dataHolder.bridge$setVanished(false);
-        }
+        dataHolder.bridge$setVanished(vanish);
+        dataHolder.bridge$setUncollideable(collision);
+        dataHolder.bridge$setUntargetable(untargetable);
```

Comment:
Okay, so I was looking about and thinking about previous discussions, and if vanish is `false`, the other two keys are removed from the user compound anyway ([see here](https://github.com/SpongePowered/SpongeCommon/blob/7eeea95c4b167d8bd728950e349e0bb9480090ba/src/main/java/org/spongepowered/common/entity/player/SpongeUser.java#L448-L451) and [here](https://github.com/SpongePowered/SpongeCommon/blob/7eeea95c4b167d8bd728950e349e0bb9480090ba/src/main/java/org/spongepowered/common/entity/player/SpongeUser.java#L469-L473)). So in the interests of getting fix in, I think we should change the calls as follows:

```suggestion
        dataHolder.bridge$setUncollideable(vanish && collision);
        dataHolder.bridge$setUntargetable(vanish && untargetable);
```

because that's clearly the intention behind what's going on - even though I'm not 100% convinced it's what we should be doing.

I would much rather have the `isUncolliable` methods renamed and updated to also consider vanish state, but I am more interested in having this fixed without having this stuck for ages because that's not good for our users.